website/project/model.py

```diff
@@ -2318,6 +2318,11 @@ def remove_contributors(self, contributors, auth=None, log=True, save=False):
                 contributor=contrib, auth=auth, log=False,
             )
             results.append(outcome)
+            if outcome:
+                for file_checked in contrib.files_checked_out():
+                    if file_checked.node._id == self._id:
```

Comment:
This can be made more efficent using a query rather than filtering in  `for` loop.